libs/blocks/global-navigation/global-navigation.js

```diff
@@ -182,6 +184,7 @@ class Gnav {
       loadBaseStyles,
       this.decorateMainNav,
       this.decorateTopNav,
+      this.decorateAside,
```

Comment:
I was looking at @mokimo's suggestion to dynamically import `decorateAside` logic, maybe we can use something like
```js
const tasks = [ loadBaseStyles, this.decorateMainNav, thisDecorateTopNav, this.decorateTopNavWrapper, ... ];
if (getMetadata('gnav-promo-source')) {
  this.decorateAside = await import('.../path');
  tasks.splice(2, 0, this.decorateAside);
}
```

Reply:
I know we would be making an additional call, but might be worth it vs. loading the code and never use it.

Reply:
I kept a hybrid approach, load minimal stuff here, but move the decoration logic to a module, https://github.com/adobecom/milo/pull/1450/commits/f82e56982923d0f4e62ad972db8f646e2a45fd68